core/test/lib/tracehouse/trace-processor-test.js

```diff
@@ -943,26 +944,51 @@ Object {
-    // The primary process events should make up more than 40% of all key trace events
-    expect(pidChangeTraceSummarized.processEventsPct).toBeGreaterThanOrEqual(0.4);
-
-    // The main frame's events should make up more than 40% of all key trace events
-    // TODO: fix!
-    // expect(pidChangeTraceSummarized.frameEventsPct).toBeGreaterThanOrEqual(0.4);
+    it('with a basic single PID trace', () => {
+      const lcpTraceSummarized = summarizeTrace(lcpTrace);
+      expect(lcpTraceSummarized.mainFramePids.size).toEqual(1);
+      // The primary process events should make up more than 40% of all key trace events
+      expect(lcpTraceSummarized.processEventsPct).toBeGreaterThanOrEqual(0.4);
+      // The main frame's events should make up more than 40% of all key trace events
+      expect(lcpTraceSummarized.frameEventsPct).toBeGreaterThanOrEqual(0.4);
+    });
+
+    it('with a multi PID trace', () => {
+      const pidChangeTraceSummarized = summarizeTrace(pidChangeTrace);
+      expect(pidChangeTraceSummarized.mainFramePids.size).toEqual(2);
+      // The primary process events should make up more than 40% of all key trace events
+      expect(pidChangeTraceSummarized.processEventsPct).toBeGreaterThanOrEqual(0.4);
+      // The main frame's events should make up more than 40% of all key trace events
+      // TODO: fix!
+      // expect(pidChangeTraceSummarized.frameEventsPct).toBeGreaterThanOrEqual(0.4);
+    });
+
+    // FrameCommittedInBrowser w/o processId, but w/ processPsuedoId, and later a ProcessReadyInBrowser
+    it('with a processPsuedoId navigation', () => {
+      // A 'normal' FrameCommittedInBrowser's data is:
+      //                 {"frame":"FRAME_ID","name":"","processId":72647,"url":"https://memegen.corp.google.com/"}
+      // But if the processId isn't ready at frame creation, we get this pair:
+      // {"args":{"data":{"frame":"FRAME_ID","name":"","processPseudoId":"0x7ff70022ca00","url":"https://memegen.com/"}},"cat":"disabled-by-default-devtools.timeline","name":"FrameCommittedInBrowser","ph":"I","pid":744,"s":"t","tid":775,"ts":123265406529,"tts":10824502153},
+      // {"args":{"data":{"frame":"FRAME_ID","processId":72647,"processPseudoId":"0x7ff70022ca00"}},"cat":"disabled-by-default-devtools.timeline","name":"ProcessReadyInBrowser","ph":"I","pid":744,"s":"t","tid":775,"ts":123265450207,"tts":10824519750},
+      const psuedoProcTrace = JSON.parse(JSON.stringify(decentlyModernTrace));
+      const fcibEvt = psuedoProcTrace.traceEvents.find(e => e.name === 'FrameCommittedInBrowser');
+      const {url, processId, frame, name} = fcibEvt.args.data;
+      expect(processId).toBeTruthy();
+      fcibEvt.args.data = {frame, name, url, processPseudoId: '0xbaabaa'};
+
+      const procReadyEvt = JSON.parse(JSON.stringify(fcibEvt));
+      procReadyEvt.name = 'ProcessReadyInBrowser';
+      procReadyEvt.args.data = {frame, processId, processPseudoId: '0xbaabaa'};
```

Comment:
Oh, do the `processPseudoId`s match? Almost feels like information we should be using in trace-processor (only look at `ProcessReadyInBrowser` events that have a `processPseudoId` that matches a `FrameCommittedInBrowser` without a `proccessId`?) but maybe it can never matter

Reply:
yah we definitely could. and I considered it. But I think it's kinda overkill.
I'm happy to revisit.